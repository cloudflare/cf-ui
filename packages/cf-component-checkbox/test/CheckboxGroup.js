const React = require('react');
const assertEqualJSX = require('assert-equal-jsx');
const CheckboxGroup = require('../src/CheckboxGroup');
const Checkbox = require('../src/Checkbox');

describe('CheckboxGroup', function() {
  it('should render', function() {
    assertEqualJSX(
      <CheckboxGroup
        values={['option1']}
        onChange={() => {}}
        options={[
          { label: 'Option 1', name: 'group-option-1', value: 'option1' },
          { label: 'Option 2', name: 'group-option-2', value: 'option2' }
        ]}
      />,
      // should equal
      <div className="cf-checkbox__group">
        <Checkbox
          label="Option 1"
          name="group-option-1"
          value="option1"
          checked={true}
          onChange={() => {}}
        />
        <Checkbox
          label="Option 2"
          name="group-option-2"
          value="option2"
          checked={false}
          onChange={() => {}}
        />
      </div>
    );
  });
});
