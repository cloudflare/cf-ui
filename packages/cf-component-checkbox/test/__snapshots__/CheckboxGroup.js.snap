// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render 1`] = `
<div
  className="cf-checkbox__group"
>
  <label
    className="cf-checkbox cf-checkbox--checked"
    htmlFor="group-option-1"
  >
    <input
      checked={true}
      className="cf-checkbox__input"
      id="group-option-1"
      name="group-option-1"
      onChange={[Function]}
      type="checkbox"
      value="option1"
    />
    <span
      className="cf-checkbox__label"
    >
      Option 1
    </span>
  </label>
  <label
    className="cf-checkbox"
    htmlFor="group-option-2"
  >
    <input
      checked={false}
      className="cf-checkbox__input"
      id="group-option-2"
      name="group-option-2"
      onChange={[Function]}
      type="checkbox"
      value="option2"
    />
    <span
      className="cf-checkbox__label"
    >
      Option 2
    </span>
  </label>
</div>
`;
